File: scripts/verify-live-connect.mjs
/**
 * Verifica conexión Gemini Live native audio (sin languageCode).
 * Uso: node scripts/verify-live-connect.mjs
 */
import { readFileSync } from "node:fs";
import { resolve, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const root = resolve(dirname(fileURLToPath(import.meta.url)), "..");
const envPath = resolve(root, "apps/api/.env");

function loadApiKey() {
  const raw = readFileSync(envPath, "utf8");
  const match = raw.match(/^GOOGLE_API_KEY=(.+)$/m);
  if (!match) throw new Error("GOOGLE_API_KEY no encontrada en apps/api/.env");
  return match[1].trim().replace(/^["']|["']$/g, "");
}

const MODEL = "gemini-2.5-flash-native-audio-preview-12-2025";

async function main() {
  const apiKey = loadApiKey();
  const { GoogleGenAI, Modality } = await import("@google/genai");

  const ai = new GoogleGenAI({
    apiKey,
    httpOptions: { apiVersion: "v1alpha" },
  });

  let setupOk = false;
  let closed = null;

  const session = await ai.live.connect({
    model: MODEL,
    config: {
      responseModalities: [Modality.AUDIO],
      speechConfig: {
        voiceConfig: { prebuiltVoiceConfig: { voiceName: "Aoede" } },
      },
      inputAudioTranscription: {},
      outputAudioTranscription: {},
    },
    callbacks: {
      onopen: () => console.log("[OK] WebSocket open"),
      onmessage: (msg) => {
        if (msg.setupComplete) {
          setupOk = true;
          console.log("[OK] setupComplete");
        }
      },
      onerror: (e) => console.error("[ERR]", e.message ?? e),
      onclose: (e) => {
        closed = { code: e?.code, reason: e?.reason };
        console.log("[CLOSE]", closed);
      },
    },
  });

  await new Promise((r) => setTimeout(r, 3000));
  try {
    session.close();
  } catch {
    /* ignore */
  }
  await new Promise((r) => setTimeout(r, 500));

  if (closed?.code === 1007) {
    console.error("FALLO: language/config", closed.reason);
    process.exit(1);
  }
  if (!setupOk) {
    console.error("FALLO: sin setupComplete");
    process.exit(1);
  }
  console.log("PASS: Live connect estable");
}

main().catch((err) => {
  console.error("FALLO:", err.message ?? err);
  process.exit(1);
});

File: apps/api/scripts/verify_live_connect.py
"""Verifica Gemini Live native audio sin language_code."""
from __future__ import annotations

import asyncio
import os
import sys
from pathlib import Path

API_ROOT = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(API_ROOT))

from dotenv import load_dotenv

load_dotenv(API_ROOT / ".env")

MODEL = "gemini-2.5-flash-native-audio-preview-12-2025"


async def main() -> int:
    api_key = os.getenv("GOOGLE_API_KEY", "").strip()
    if not api_key:
        print("FALLO: GOOGLE_API_KEY vacía")
        return 1

    from google import genai
    from google.genai import types

    client = genai.Client(
        api_key=api_key,
        http_options=types.HttpOptions(api_version="v1alpha"),
    )

    print(f"Modelo: {MODEL}")
    try:
        async with client.aio.live.connect(
            model=MODEL,
            config=types.LiveConnectConfig(
                response_modalities=[types.Modality.AUDIO],
                speech_config=types.SpeechConfig(
                    voice_config=types.VoiceConfig(
                        prebuilt_voice_config=types.PrebuiltVoiceConfig(
                            voice_name="Charon",
                        ),
                    ),
                ),
                input_audio_transcription=types.AudioTranscriptionConfig(),
                output_audio_transcription=types.AudioTranscriptionConfig(),
            ),
        ) as session:
            print("[OK] WebSocket open")
            await asyncio.sleep(2)
            if session.setup_complete is None:
                print("FALLO: sin setupComplete")
                return 1
            sid = session.setup_complete.session_id
            print(f"[OK] setupComplete session_id={sid}")
    except Exception as exc:
        msg = str(exc)
        if "Unsupported language" in msg or "language" in msg.lower():
            print(f"FALLO language config: {msg}")
        else:
            print(f"FALLO: {msg}")
        return 1

    print("PASS: Live connect estable sin language_code")
    return 0


if __name__ == "__main__":
    raise SystemExit(asyncio.run(main()))
